administrator/components/com_associations/models/fields/associatedcomponent.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_associations
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+defined('JPATH_BASE') or die;
+JFormHelper::loadFieldClass('groupedlist');
+/**
+ * A drop down containing all valid HTTP 1.1 response codes.
+ *
+ * @package     Joomla.Administrator
+ * @subpackage  com_redirect
+ * @since       __DEPLOY_VERSION__
+ */
+class JFormFieldAssociatedComponent extends JFormFieldGroupedList
+{
+	/**
+	 * The form field type.
+	 *
+	 * @var    string
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $type = 'AssociatedComponent';
+
+	protected $articles = array(
+		'com_content.articles' => 'Articles',
+		'com_content.categories' => 'Categories',
+	);
+	
+	/**
+	 * Method to get the field input markup.
+	 *
+	 * @return  string	The field input markup.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getGroups()
```

Comment:
Also as you want to get the list of components items
to check if associations are implemented for items (NOT categories), you should rather use
`escapeshellarg("associationsContext")` 
instead of
`escapeshellarg("JLanguageAssociations::isEnabled()")` as this last one will also list com_categories.

As we discussed in the chat, to check for the existence of categories in a specific component, it looks simpler to do it in the frontend helper for each component.

```
Tue, Jul 5 11:26 AM

We need an association helper in the component frontend (helpers/association.php) where one looks for the views in the getAssociations method


abstract class ContactHelperAssociation extends CategoryHelperAssociation

and then the code depends on the available views
if ($view == 'contact')
etc.

and, further:

if ($view == 'category'  
 
$view == 'categories') { return self::getCategoryAssociations($id, 'com_contact'); } 

therefore, we need indeed to check that the class contains extends CategoryHelperAssociation but also make sure that the method getAssociations() contains 
return self::getCategoryAssociations()
```